examples/stepper/stepper.c

```diff
@@ -0,0 +1,151 @@
+/****************************************************************************
+ * apps/examples/stepper/stepper.c
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.  The
+ * ASF licenses this file to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ *
+ ****************************************************************************/
+
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <sys/ioctl.h>
+#include <fcntl.h>
+#include <stdio.h>
+#include <stdbool.h>
+
+#include <nuttx/motor/stepper.h>
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+struct stepper_args
+{
+  FAR const char * path;
+  int microstep;
+  int speed;
+  int steps;
+};
+
+static int parse_args(int argc, FAR char *argv[],
+                      FAR struct stepper_args *args)
+{
+  if (argc < 3)
+    {
+      return -1;
+    }
+
+  args->path = argv[1];
+  args->microstep = -1;
+  args->speed = 200;
+
+  for (int i = 2; i < argc; ++i)
```

Comment:
Please use C89, don't declare "i" in the for loop, declare it at begging of the function.

Reply:
My bad, old habits